lib-multisrc/libgroup/src/eu/kanade/tachiyomi/multisrc/libgroup/LibGroup.kt

```diff
@@ -18,284 +22,247 @@ import eu.kanade.tachiyomi.source.model.Page
-            add("x-csrf-token", csrfToken)
+    private var _constants: Constants? = null
+    private fun getConstants(): Constants {
+        if (_constants == null) {
+            try {
+                _constants = client.newCall(
+                    GET("https://api.$apiDomain/api/constants?fields[]=genres&fields[]=tags&fields[]=types&fields[]=scanlateStatus&fields[]=status&fields[]=format&fields[]=ageRestriction&fields[]=imageServers", headers),
+                ).execute().parseAs<Data<Constants>>().data
+                return _constants!!
+            } catch (ex: SerializationException) {
+                throw Exception("Ошибка сериализации. Проверьте сайт.")
+            }
         }
-        .build()
+        return _constants!!
+    }
 
-    // Latest
-    override fun latestUpdatesRequest(page: Int) = throw UnsupportedOperationException() // popularMangaRequest()
-    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> {
-        if (csrfToken.isEmpty()) {
-            return client.newCall(popularMangaRequest(page))
-                .asObservableSuccess()
-                .flatMap { response ->
-                    // Obtain token
-                    val resBody = response.body.string()
-                    csrfToken = "_token\" content=\"(.*)\"".toRegex().find(resBody)!!.groups[1]!!.value
-                    return@flatMap fetchLatestMangaFromApi(page)
+    private fun checkForToken(chain: Interceptor.Chain): Response {
+        val req = chain.request().newBuilder()
+        val url = chain.request().url.toString()
+        if (url.contains("api.$apiDomain") && !url.contains("/api/auth/me")) {
+            if (bearerToken.isNullOrBlank()) {
+                bearerToken = loadToken()
+            }
+            if (bearerToken != "none") {
+                req.apply {
+                    addHeader("Authorization", bearerToken.orEmpty())
                 }
+            }
         }
-        return fetchLatestMangaFromApi(page)
+        return chain.proceed(req.build())
     }
 
-    private fun fetchLatestMangaFromApi(page: Int): Observable<MangasPage> {
-        return client.newCall(POST("$baseUrl/filterlist?dir=desc&sort=last_chapter_at&page=$page&chapters[min]=1", catalogHeaders()))
-            .asObservableSuccess()
-            .map { response ->
-                latestUpdatesParse(response)
+    @SuppressLint("ApplySharedPref")
+    private fun loadToken(): String {
+        try {
+            val token = preferences.getString(TOKEN_STORE, "")!!.parseAs<AuthToken>()
+            if (token.isExpired() || !isUserTokenValid(token.getToken())) {
+                val refreshedToken: AuthToken? = refreshToken()
+                if (refreshedToken != null) {
+                    val str = json.encodeToString(refreshedToken)
+                    preferences.edit().putString(TOKEN_STORE, str).commit()
+                }
+            }
+            return token.getToken()
```

Comment:
why return the old token after refreshing it?

Reply:
Indeed forgot to override it